React/CSS-in-JS/Chakra.md

```diff
@@ -0,0 +1,498 @@
+
+Components styles are defined in `theme/components/<component-name>.ts`
+Each component style will export these objects:
+
+- `baseStyles`
+- `sizes`
+- `variants`
+- `defaultProps`
+
+`baseStyle` are styles that all button types share.
+
+```tsx
+const baseStyle: StyleObjectOrFn = {
+  lineHeight: "1.2",
+  borderRadius: "md",
+  _focus: {
+    boxShadow: "outline",
+  },
+  _disabled: {
+    opacity: 0.4,
+    cursor: "not-allowed",
+    boxShadow: "none",
+  },
+  _hover: {
+    cursor: "pointer",
+  },
+};
+```
+
+<!-- TODO change this example to use color schema -->
+
+`variants` represents visual style.
+
+```tsx
+const variants: { [variant: string]: StyleObjectOrFn } = {
+  solid: {
+    bg: 'primary.100',,
+    color: 'white',
+    _hover: {
+      bg: 'primary.400',
+    },
+    _disabled: {
+      bg: 'neutral.200',
+    }
+  },
+	outline: (props: Dict): SystemStyleObject => ({
+    bg: 'transparent',,
+    border: '1px solid',
+    borderColor: 'primary.100',
+    _hover: {
+      bg: 'primary.100',
+    }
+  }),
+};
+```
+
+`sizes` determine all the component sizes like `width`, `height`, `font-size` etc
+
+```tsx
+const sizes: { [size: string]: StyleObjectOrFn } = {
+  sm: {
+    h: 8,
+    minW: 8,
+    fontSize: "sm",
+    px: 3,
+  },
+  xs: (props: Dict): SystemStyleObject => ({
+    h: 6,
+    minW: 6,
+    fontSize: "xs",
+    px: 2,
+  }),
+};
+```
+
+Naming used for sizes:
+
+- `lg`
+- `md`
+- `sm`
+- `xs`
+
+`defaultProps` are default values for `size` and `variant`
+
+```tsx
+const defaultProps = {
+  variant: "solid",
+  size: "md",
+};
+```
+
+For more info about custom component styles read [docs](https://chakra-ui.com/docs/theming/customize-theme#customizing-component-styles)
+
+### Color mode
+
+When you use the ChakraProvider at the root of your app, you can automatically use color mode in your apps. By default, most of Chakra UI component are dark mode compatible. To handle color mode manually in your application, use the `useColorMode` or `useColorModeValue` hooks.
+
+> Tip: Chakra stores the color mode in localStorage and uses CSS variables to ensure the color mode is persistent.
+
+#### useColorMode
+
+`useColorMode` is a React hook that gives you access to the current color mode, and a function to toggle the color mode.
+
+```tsx
+function Example() {
+  const { colorMode, toggleColorMode } = useColorMode();
+  return (
+    <header>
+      <Button onClick={toggleColorMode}>
+        Toggle {colorMode === "light" ? "Dark" : "Light"}
+      </Button>
+    </header>
+  );
+}
+```
+
+#### useColorModeValue
+
+`useColorModeValue` is React hook that takes 2 arguments, first is value for light mode and second is value for dark mode, and returnees the the value based on the active color mode
+
+```tsx
+const value = useColorModeValue(lightModeValue, darkModeValue);
+```
+
```

Comment:
Maybe we could add one more section dedicated to the chakra factory https://chakra-ui.com/docs/features/chakra-factory.
It could be used similar to emotion `styled` component but it will be aware of the theme props.
```
const Card = chakra("div", {
  baseStyle: {
    shadow: "lg",
    rounded: "lg",
    bg: "white",
  },
})
```

Reply:
I added chakra factory section + example with chakra factory function and `react-datepicker`, similar to there's `react-input-autoresize` [example](https://chakra-ui.com/docs/features/chakra-factory#chakra-factory-function)